---
title: "NanoStringFeb17"
author: "Luke Hayden"
date: "8th March 2017"
output: html_document
---
Second round of analyses on the February 2017 Nanostring dataset, combining old and new analyses. Here, I have done some more in-depth analyses on the variability of each marker (coefficients of variation and their effects on age signal), ranked markers by testing each for a significant old vs young difference, applied this test to improve old vs young signal, examined how regeneration affects these markers and examined the effect of normalisation on these results. 


file:///C:/Users/Luke/Documents/nstringnov17/20170208_lh 8-2-17_lh 8 2 17_12.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170207_luke 7 2 17_lh 7 2 17_01.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170207_luke 7 2 17_lh 7 2 17_02.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170207_luke 7 2 17_lh 7 2 17_03.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170207_luke 7 2 17_lh 7 2 17_04.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170207_luke 7 2 17_lh 7 2 17_05.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170207_luke 7 2 17_lh 7 2 17_06.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170207_luke 7 2 17_lh 7 2 17_07.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170207_luke 7 2 17_lh 7 2 17_08.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170207_luke 7 2 17_lh 7 2 17_09.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170207_luke 7 2 17_lh 7 2 17_10.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170207_luke 7 2 17_lh 7 2 17_11.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170207_luke 7 2 17_lh 7 2 17_12.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170208_lh 8-2-17_lh 8 2 17_01.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170208_lh 8-2-17_lh 8 2 17_02.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170208_lh 8-2-17_lh 8 2 17_03.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170208_lh 8-2-17_lh 8 2 17_04.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170208_lh 8-2-17_lh 8 2 17_05.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170208_lh 8-2-17_lh 8 2 17_06.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170208_lh 8-2-17_lh 8 2 17_07.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170208_lh 8-2-17_lh 8 2 17_08.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170208_lh 8-2-17_lh 8 2 17_09.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170208_lh 8-2-17_lh 8 2 17_10.RCC
file:///C:/Users/Luke/Documents/nstringnov17/20170208_lh 8-2-17_lh 8 2 17_11.RCC

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
#source("https://bioconductor.org/biocLite.R")
#biocLite("DESeq2")
#source("https://bioconductor.org/biocLite.R")
#biocLite("DESeq2")

library(ggplot2)
library(DESeq2)
library(rmarkdown)
library(pracma)
library(gplots)
library(RColorBrewer)
library(reshape2)
library(scales)
library(ggbiplot)
library(gtools)
library(devtools)
library(NanoStringNorm)
library(ggthemes)
library(grid)
library(gridExtra)
library(sjstats)


multiplot <- function(..., plotlist = NULL, file, cols = 1, layout = NULL) {
  require(grid)
  
  plots <- c(list(...), plotlist)
  
  numPlots = length(plots)
  
  if (is.null(layout)) {
    layout <- matrix(seq(1, cols * ceiling(numPlots/cols)),
                     ncol = cols, nrow = ceiling(numPlots/cols))
  }
  
  if (numPlots == 1) {
    print(plots[[1]])
    
  } else {
    grid.newpage()
    pushViewport(viewport(layout = grid.layout(nrow(layout), ncol(layout))))
    
    for (i in 1:numPlots) {
      matchidx <- as.data.frame(which(layout == i, arr.ind = TRUE))
      
      print(plots[[i]], vp = viewport(layout.pos.row = matchidx$row,
                                      layout.pos.col = matchidx$col))
    }
  }
}
```



```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}


gm <- read.csv("gmark.csv")

setwd("~/nstringnov17/ctg5and6")
ct5and6.raw <- read.markup.RCC()


setwd("~/nstringnov17/ctg5hi")
ct5and6hi.raw <- read.markup.RCC()


setwd("~/nstringnov17/ctg7and8")
ct7and8.raw <- read.markup.RCC()

setwd("~/nstringnov17/ctg3and4")

ct3and4.raw <- read.markup.RCC()

````

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}
codeset <- read.table("CodeSet.txt")
codeset$shortname <- substr(codeset$Customer_Identifier, 1,2)
Cht <- read.table("CodeSetCountdata.txt")


ctg5.norm <- NanoStringNorm(x = ct5.raw, anno = NA, CodeCount ='sum', Background ="mean", SampleContent ='housekeeping.sum', round.values = FALSE, take.log = FALSE,return.matrix.of.endogenous.probes = TRUE);


ctg5and6.norm <- NanoStringNorm(x = ct5and6.raw, anno = NA, CodeCount ='sum', Background ="mean", SampleContent ='housekeeping.sum', round.values = FALSE, take.log = FALSE,return.matrix.of.endogenous.probes = TRUE);


ctg5and6hi.norm <- NanoStringNorm(x = ct5and6hi.raw, anno = NA, CodeCount ='sum', Background ="mean", SampleContent ='housekeeping.sum', round.values = FALSE, take.log = FALSE,return.matrix.of.endogenous.probes = TRUE);

ctg7and8.norm <- NanoStringNorm(x = ct7and8.raw, anno = NA, CodeCount ='sum', Background ="mean", SampleContent ='housekeeping.sum', round.values = FALSE, take.log = FALSE,return.matrix.of.endogenous.probes = TRUE);

ctg7and8hi.norm <- NanoStringNorm(x = ct7and8hi.raw, anno = NA, CodeCount ='sum', Background ="mean", SampleContent ='housekeeping.sum', round.values = FALSE, take.log = FALSE,return.matrix.of.endogenous.probes = TRUE);

ctg3and4.norm <- NanoStringNorm(x = ct3and4.raw, anno = NA, CodeCount ='sum', Background ="mean", SampleContent ='housekeeping.sum', round.values = FALSE, take.log = FALSE,return.matrix.of.endogenous.probes = TRUE);


colnames(ctg5.norm) <- c("SFPA", "SFPB", "LFPA", "LFPB", "SMPA", "SMPB", "LMPA", "LMPB", "LMIA", "LMIB", "LMIC", "LMID")

colnames(ctg5and6.norm) <- c("SFPA", "SFPB", "LFPA", "LFPB", "SMPA", "SMPB", "LMPA", "LMPB", "LMIA", "LMIB", "LMIC", "LMID", "Abef", "Aun", "Areg", "Bbef", "Bun", "Breg", "Cbef", "Cun", "Creg", "Dbef", "Dun", "Dreg")

colnames(ctg5and6hi.norm) <-  c("SFPA", "SFPB", "LFPA", "LFPB", "SMPA", "SMPB", "LMPA", "LMPB", "LMIA", "LMIB", "LMIC", "LMID", "Abef", "Aun", "Areg", "Bbef", "Bun", "Breg", "Cbef", "Cun", "Creg", "Dbef", "Dun", "Dreg")


colnames(ctg7and8.norm) <- c("SFPC","SFPD", "LFPC", "LFPD", "SMPC", "SMPD","LMPC", "LMPD", "0mPA", "0mPB", "10mPA", "20mPA", "Ebef", "Eun", "Ereg", "Fbef", "Fun", "Freg", "Gbef", "Gun", "Greg", "LFIA", "LFIB", "20mPB")

colnames(ctg7and8hi.norm) <-c("SFPC","SFPD", "LFPC", "LFPD", "SMPC", "SMPD","LMPC", "LMPD", "0mPA", "0mPB", "10mPA", "20mPA", "Ebef", "Eun", "Ereg", "Fbef", "Fun", "Freg", "Gbef", "Gun", "Greg", "LFIA", "LFIB", "20mPB")


colnames(ctg3and4.norm) <- c("O1a", "O2a", "Y1a", "Y2a", "OR1a", "OR2a", "YR1a", "YR2a", "FY5a", "FY6a", "FO5a", "FO7a", "O1b", "O2b", "Y1b", "Y2b", "OR1b", "OR2b", "YR1b", "YR2b", "FY7", "FY8", "FO8", "FO9")

ctg3to8 <- cbind(ctg3and4.norm, ctg5and6hi.norm, ctg7and8.norm)
sampleinfo <- data.frame(sample=colnames(ctg3to8), sex=c(rep("F", 28), rep("M", 8), rep("F", 16), rep("M", 8), rep("F", 12)), age=c(rep(c("O", "O", "Y", "Y", "O", "O", "Y", "Y","Y", "Y", "O", "O"), 2), rep(c("Y", "Y", "O", "O", "Y", "Y", "O", "O", "O", "O", "O", "O", rep("O", 12)), 2)), reg=c(rep(c(rep("", 4), rep("R", 4), rep("", 4)), 2), c(rep("", 12), rep(c("", "", "R"),4 )   ), c(rep("", 12), rep(c("", "", "R"),3 )   ), "", "", "") , ctg=c(rep("3", 12), rep("4", 12), rep("5", 12), rep("6", 12), rep("7", 12), rep("8", 12))  )
sampleinfo$type <- paste(sampleinfo$age, sampleinfo$reg, sep="")
sampleinfo$qual <- "ok"
sampleinfo$qual[c(1,7,8, 25, 43, 52)] <- "bad"

```

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}
#all <- ctg3to8
#all$isg <- rownames(ctg3to8) %in% gm
mc36 <- melt(ctg3to8)
mc36$exp <- mc36$value > 1
mc36$isg <- mc36$Var1 %in% gm$marker
ggplot(mc36, aes(y=value, x=Var1, colour=isg)) +geom_point(alpha=0.3) +facet_wrap(~Var2) +scale_y_log10() +scale_colour_manual


#ggplot(mc36, aes(y=value, x=Var1, colour=exp)) +geom_point(alpha=0.3, shape=95, size=5) +facet_wrap(~Var2) +scale_y_log10() + theme(  axis.text.x = element_blank())

```


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
mc5 <- melt(ctg5and6.norm)
mc5h <- melt(ctg5and6hi.norm)
mc5h$hival <- mc5$value

ggplot(data=mc5h, aes(x=value, y=hival, colour=Var2)) +geom_point() +facet_wrap(~Var2, ncol=6) +scale_x_log10() +scale_y_log10()
```



```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
#bcmin <- all.norm[,c(1:4)]
ctg5h.norm <- ctg5hi.norm
colnames(ctg5h.norm) <- paste(colnames(ctg5hi.norm),"h")
bcmin <- cbind(ctg5.norm, ctg5h.norm)
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c(rep(c("Y", "Y", "O","O","Y", "Y", "O","O","O","O","O","O"),2))
#bc.groups <- c(rep("low",12), rep("high", 12))

#bc.groups <- c("Y", "Y", "O","O")

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 195 markers")+ scale_colour_manual(values=c("cornflower blue", "red3"))

```

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
#bcmin <- all.norm[,c(1:4)]
bcmin <- ctg3to6
bcmin <- subset(bcmin,rownames(bcmin) %in% gm$marker)
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
#bc.groups <- c("Y", "Y", "O","O","Y", "Y", "O","O","O","O","O","O")

#bc.groups <- c("Y", "Y", "O","O")

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc))+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 195 markers")+ scale_colour_manual(values=c("cornflower blue", "red3"))


```



```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
#bcmin <- all.norm[,c(1:4)]
bcmin <- ctg5and6.norm
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O","O", "OR", "O","O", "OR", "O","O", "OR", "O","O", "OR", "Y", "Y", "O","O","Y", "Y", "O","O","O","O","O","O" )                 


#bc.groups <- c("Y", "Y", "O","O")

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 195 markers")+ scale_colour_manual(values=c("cornflower blue", "forest green", "red3"))

```

And now if we use l2fc in all old vs young of our new samples as the criterion, we should expect to see somewhat better separation. However, this is somewhat circular as we are choosing genes on the basis of them differing in expression between the two groups

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

bcmin <- ctg3and4.norm[,c(2,3,4,9,10,11,12,13:16,21:24)]
bcmin <- subset(bcmin, rownames(bcmin) %in% gm$marker)
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y")+ scale_colour_manual(values=c("cornflower blue", "red3"))


rg <-subset(ctg5.norm, rownames(ctg5.norm) %in% gm$marker)

rg.sc <- scale(t(rg), center= bc.pca$center)
rg.pred <- rg.sc %*% bc.pca$rotation 


rgd.pca <- bc.pca
rgd.pca$x <- rbind(bc.pca$x, rg.pred)
                   
                   
rgd.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O", "Y", "Y", "O","O","Y", "Y", "O","O","O","O","O","O")                 

ggbiplot(rgd.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(rgd.pca$x), groups=rgd.groups, varname.abbrev = TRUE)+ theme(legend.direction = 'horizontal', legend.position = 'top')+ggtitle("PCA Old vs Young, using just our subset of 61 markers + new projected")+ scale_colour_manual(values=c("cornflower blue","red3"))


```


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}




bcmin <- ctg3to8[,c(2,3,4,9,10,11,12,13:16,21:24,26:28,49:52)]
bcmin <- subset(bcmin, rownames(bcmin) %in% gm$marker)
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c( "O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O","Y","O","O","Y", "Y","O","O")

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 61 chosen markers")+ scale_colour_manual(values=c("cornflower blue", "red3"))



````


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

log <- sampleinfo$sex =="F" & sampleinfo$ctg %in% c(3,4,5,7) & sampleinfo$qual == "ok" & sampleinfo$reg == ""


bcmin <- ctg3to8[,log== TRUE]
bcmin.groups <- subset(sampleinfo$type, log==TRUE)
bcmin <- subset(bcmin, rownames(bcmin) %in% gm$marker)
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- bcmin.groups

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 61 chosen markers")+ scale_colour_manual(values=c("cornflower blue", "red3"))



rg <-ctg3to8[,c(5,6,17,18,37:42,44:48,61:69)]
rg <- subset(rg, rownames(rg) %in% gm$marker)
rg.groups <- sampleinfo$type[c(5,6,17,18,37:42,44:48,61:69)]

rg.sc <- scale(t(rg), center= bc.pca$center)
rg.pred <- rg.sc %*% bc.pca$rotation 


rgd.pca <- bc.pca
rgd.pca$x <- rbind(bc.pca$x, rg.pred)
                   
                   
rgd.groups <- append(bcmin.groups, rg.groups)                

ggbiplot(rgd.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(rgd.pca$x), groups=rgd.groups, varname.abbrev = TRUE)+ theme(legend.direction = 'horizontal', legend.position = 'top')+ggtitle("PCA Old vs Young, using just our subset of 61 markers + new projected")+ scale_colour_manual(values=c("cornflower blue", "forest green", "red3"))

````


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
an <- ctg3to8/rowMax(ctg3to8)
log <- sampleinfo$sex =="F" &  sampleinfo$qual == "ok" & sampleinfo$type != "YR"
anf <- an[,log==TRUE]
sif <- subset(sampleinfo, log)
manf <- melt(t(rbind(anf, t(sif))), id.vars=colnames(sif))


```


#Age effect by marker
Still using the same samples: old vs young from the recent dataset only. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

mndbcg$type <- paste(mndbcg$age, " , ", mndbcg$dir)
mndbcg <- mndbcg[order(mndbcg$dir),]
mndbcg$mkty <- paste(substr(mndbcg$dir, 1, 2)," ", substr(mndbcg$marker, 1, 2))


ggplot(mndbcg, aes(x=mkty, y=value, color=age, fill=dir)) +geom_point(shape=95, size=4.5) +ggtitle("61 chosen markers, normalised by mean expression")+scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young"))+ylab("Nanostring expression (normalised)") +xlab("Sample")+ theme(axis.text.x = element_text(angle = 60, hjust = 1))+stat_summary(fun.y=mean, geom="point", shape=21)+ scale_fill_manual(values=c("white", "azure4"), name="Direction")+geom_hline(yintercept = 1, linetype=2) +geom_vline(xintercept="moo") + scale_y_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8) ) +scale_x_discrete(labels=substr(ndbcg$marker,1,2))


ggplot(mndbcg, aes(x=mkty, y=value, color=age)) +geom_boxplot(aes(fill=dir)) +ggtitle("61 chosen markers, normalised by mean expression")+scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young"))+ylab("Nanostring expression (normalised)") +xlab("Sample")+ theme(axis.text.x = element_text(angle = 60, hjust = 1))+ scale_fill_manual(values=c("white", "azure4"), name="Direction")+geom_hline(yintercept = 1, linetype=2) +geom_vline(xintercept="moo")+ scale_y_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8) )+scale_x_discrete(labels=substr(ndbcg$marker,1,2))



```


Each marker is not 100% predictive, some old samples mix among the young ones. 


#Coefficient of variation vs Old vs Young t-test


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
mkc$cv <- cvvec

ggplot(mkc, aes(y=p_value, x=cv, colour=sig)) +geom_point() +stat_smooth(aes(y=p_value, x=cv), method=lm, se=FALSE, inherit.aes=FALSE) +ylim(0,1) +scale_colour_manual(labels=c("p<0.05", "p>0.05"), name="t-test result", values=c("black", "forest green")) +ggtitle("Does Coefficient of Variation predict old vs young change?") +xlab("Coefficient of Variation (Nanostring vs RNAseq)") +ylab("Old vs Young t-test p-value")



```

#Marker Identity
Drosophila top BLAST hits:
```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
redcs <- subset(codeset, sig == TRUE)
print(redcs$DrosoTopHit[order(redcs$DrosoTopHit)])
#print(redcs$TopHit)

```

#Project regenerated samples onto PCA of Old vs Young

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
bcg <- subset(bc, sig ==TRUE)
bcg <- bcg[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcg)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
rd.pca <- bc.pca
rd<- bcg
rg <-subset(bc, sig ==TRUE)
rg <- rg[,c(5:6,17:20)]


rg.sc <- scale(t(rg), center= rd.pca$center)
rg.pred <- rg.sc %*% rd.pca$rotation 


rgd.pca <- rd.pca
rgd.pca$x <- rbind(rd.pca$x, rg.pred)
                   
                   
rgd.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O", "OR", "OR", "OR", "OR", "YR", "YR")                 

ggbiplot(rgd.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(rgd.pca$x), groups=rgd.groups, varname.abbrev = TRUE)+ theme(legend.direction = 'horizontal', legend.position = 'top')+ggtitle("PCA Old vs Young, using just our subset of 61 markers + Regenerated samples projected")+ scale_colour_manual(values=c("cornflower blue","forest green", "red3",  "black"))


```

According to this, the regenerated samples are intermediate between the old and young samples. TYhis makes sense to me, as the OR samples being "younger" than the O samples fits the preliminary data, while the YR samples are three months older than the Y. 

#Marker expression levels in regenerated samples
But do these markers respond homogenously to regeneration? Lets look at how each marker responds to regeneration. 
We can plot the expression levels of each marker (normalised in relation to mean expression in old vs young samples) in old, young, regenerated young and regenerated old samples. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

mkcr <- cbind(mkc, bc[,c(5:6,17:20)]) 
nmkcr <- as.data.frame(mkcr[,c(1:15,32:37)]/rowMeans(as.matrix(mkcr[,c(1:15,32:37)])))
nmkcr <- cbind(as.data.frame(nmkcr), as.data.frame(mkc[,c(16:31)]))
nmkcr$rgl2fc <- foldchange2logratio(foldchange(rowMeans(mkcr[,c(1,6:9,14:15)]), rowMeans(mkcr[,c(32:35)])))
nmkcr$rgl2fc[nmkcr$dir == "upregulated with age"] <- nmkcr$rgl2fc[nmkcr$dir == "upregulated with age"]*-1
nmkcr$rgup <- nmkcr$rgl2fc >0
mnmkcr <- melt(nmkcr, id.vars=c(colnames(mkc[,c(16:31)]), "rgl2fc", "rgup"))
mnmkcr$class <- "Young"
mnmkcr$class[substr(mnmkcr$variable, 1,1) == "O"] <- "Old"
mnmkcr$class[substr(mnmkcr$variable, 1,2) == "OR"] <- "Old Regenerated"
mnmkcr$class[substr(mnmkcr$variable, 1,2) == "YR"] <- "Young Regenerated"
dm <- subset(mnmkcr, mnmkcr$p_value <0.05)
dm$dnm <- paste(substr(dm$dir, 1,2), dm$name)
l <- c("Younger with regen", "Older with regen")

ggplot(dm, aes(x=dnm, y=value, color=class, fill=dir)) +geom_point(shape=95, size=4.5)+stat_summary(fun.y=mean, geom="point", shape=21)+ scale_fill_manual(values=c("white", "azure4"), name="Direction")+ scale_y_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8) )+geom_hline(yintercept = 1, linetype=2)+scale_color_manual(values=c("cornflower blue","forest green", "red3",  "black"), name= "Age and Regeneration")+ theme(axis.text.x = element_text(angle = 60, hjust = 1))+ggtitle("61 chosen markers, normalised by mean expression")+ylab("Nanostring expression (normalised)") +xlab("Marker") +scale_x_discrete(labels=substr(dm$dnm, 4,5))
```

Some markers are upregulated with regeneration, some are downregulated with regeneration, irrespective of whether they are upregulated or downregulated with age. To see this more clearly, we will split those that move towards the young side (upper panel) from those that move towards the old side with regeneration. 


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

ggplot(dm, aes(x=dnm, y=value, color=class, fill=dir)) +geom_point(shape=95, size=4.5)+stat_summary(fun.y=mean, geom="point", shape=21)+ scale_fill_manual(values=c("white", "azure4"), name="Direction")+ scale_y_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8) )+geom_hline(yintercept = 1, linetype=2)+scale_color_manual(values=c("cornflower blue","forest green", "red3",  "black"), name= "Age and Regeneration")+ theme(axis.text.x = element_text(angle = 60, hjust = 1))+ggtitle("61 chosen markers, normalised by mean expression")+ylab("Nanostring expression (normalised)") +xlab("Marker")  +facet_wrap(~rgup, scales="free_x", ncol=1) + theme(strip.background = element_blank(), strip.text.x = element_blank())


```


So we have two sets of markers: those where the regenerated samples move towards the young condition (34 markers) and those where they move towards the old (27 markers)


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
#ggplot(subset(nmkcr, nmkcr$p_value <0.05), aes(x=name, y=rgl2fc)) +geom_point()
#ggplot(subset(nmkcr, nmkcr$p_value <0.05), aes(x=absnl2fc, y=rgl2fc)) +geom_point()

nmkcr$rgl2fc <- foldchange2logratio(foldchange(rowMeans(mkcr[,c(1,6:9,14:15)]), rowMeans(mkcr[,c(32:35)])))

#ggplot(nmkcr, aes(x=rgl2fc, fill=dir)) +geom_histogram(alpha=0.5) +scale_fill_manual(values=c())+theme(legend.position = "bottom", legend.title = element_blank()) +xlab("Log2Foldchange in expression in old vs regenerated old") +scale_fill_manual(name="x", values=c("darkcyan", "orangered")) 

#ggplot(nmkcr, aes(x=rgl2fc, fill=dir)) +geom_density(alpha=0.5) +scale_fill_manual(values=c())+theme(legend.position = "bottom", legend.title = element_blank()) +xlab("Log2Foldchange in expression in old vs regenerated old") +scale_fill_manual(name="x", values=c("darkcyan", "orangered")) 

```

Lets look at these on separate panels:

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

p<- ggplot(dm, aes(x=dnm, y=value, color=class, fill=dir)) +geom_point(shape=95, size=4.5)+theme_bw()+stat_summary(fun.y=mean, geom="point", shape=21)+ scale_fill_manual(values=c("white", "azure4"), name="Direction")+ scale_y_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8) )+scale_color_manual(values=c("cornflower blue","forest green", "red3",  "black"), name= "Age and Regeneration")+ theme(axis.text.x = element_text(angle = 60, hjust = 1))+ggtitle("61 chosen markers, normalised by mean expression")+ylab("Nanostring expression (normalised)") +xlab("Marker")  +facet_wrap(~class, scales="free_x", ncol=1) +geom_boxplot()+geom_hline(yintercept = 1, linetype=2)+scale_x_discrete(labels=substr(dm$dnm, 4,5)) 


ggsave(plot=p,height=10,width=10,dpi=200, filename=paste("t.pdf"), useDingbats=FALSE, limitsize = FALSE)


ggplot(dm, aes(x=dnm, y=value, color=class, fill=dir)) +geom_point(shape=95, size=4.5)+stat_summary(fun.y=mean, geom="point", shape=21)+ scale_fill_manual(values=c("white", "azure4"), name="Direction")+ scale_y_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8) )+scale_color_manual(values=c("cornflower blue","forest green", "red3",  "black"), name= "Age and Regeneration")+ theme(axis.text.x = element_text(angle = 60, hjust = 1))+ggtitle("61 chosen markers, normalised by mean expression")+ylab("Nanostring expression (normalised)") +xlab("Marker")  +facet_wrap(~class, scales="free_x", ncol=1) +geom_hline(yintercept = 1, linetype=2)+scale_x_discrete(labels=substr(dm$dnm, 4,5))

```

#Regeneration effect 

The old regenerated samples are much more variable than either the old or the young. Previously, we had split our markers into two groups: those where the regenerated samples move towards the young condition (34 markers) and those where they move towards the old (27 markers). Therefore, we might suspect that performing PCAs using these markers for young vs old will return different results. 


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

#nmkcr$rgup == FALSE & nmkcr$p_value <0.05)



bcg <- subset(bc, sig ==TRUE &nmkcr$rgup == FALSE)
bcg <- bcg[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcg)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
rd.pca <- bc.pca
rd<- bcg
rg <-subset(bc, sig ==TRUE &nmkcr$rgup == FALSE)
rg <- rg[,c(5:6,17:20)]


rg.sc <- scale(t(rg), center= rd.pca$center)
rg.pred <- rg.sc %*% rd.pca$rotation 


rgd.pca <- rd.pca
rgd.pca$x <- rbind(rd.pca$x, rg.pred)
                   
                   
rgd.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O", "OR", "OR", "OR", "OR", "YR", "YR")                 

pa <- ggbiplot(rgd.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(rgd.pca$x), groups=rgd.groups, varname.abbrev = TRUE)+ theme(legend.direction = 'horizontal', legend.position = 'top')+ggtitle("PCA Old vs Young, 34 markers + Regenerated samples projected")+ scale_colour_manual(values=c("cornflower blue","forest green", "red3",  "black"))



bcg <- subset(bc, sig ==TRUE &nmkcr$rgup == TRUE)
bcg <- bcg[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcg)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
rd.pca <- bc.pca
rd<- bcg
rg <-subset(bc, sig ==TRUE &nmkcr$rgup == TRUE)
rg <- rg[,c(5:6,17:20)]


rg.sc <- scale(t(rg), center= rd.pca$center)
rg.pred <- rg.sc %*% rd.pca$rotation 


rgd.pca <- rd.pca
rgd.pca$x <- rbind(rd.pca$x, rg.pred)
                   
                   
rgd.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O", "OR", "OR", "OR", "OR", "YR", "YR")                 

pb<- ggbiplot(rgd.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(rgd.pca$x), groups=rgd.groups, varname.abbrev = TRUE)+ theme(legend.direction = 'horizontal', legend.position = 'top')+ggtitle("PCA Old vs Young, 27 markers + Regenerated samples projected")+ scale_colour_manual(values=c("cornflower blue","forest green", "red3",  "black"))


multiplot(pa, pb, cols=1)
```



#Does regeneration have the same effect on old as on young?
Is logfoldchange in expression between old vs old regenerated correlated with logfoldchange in expression between young and young regenerated? 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

nmkcr$rgyl2fc <- foldchange2logratio(foldchange(rowMeans(mkcr[,c(2:5,10:13)]), rowMeans(mkcr[,c(36:37)])))
dn <- subset(nmkcr, nmkcr$p_value <0.05)

ggplot(dn, aes(x=rgyl2fc, y=rgl2fc, colour=dir)) +geom_point() +stat_smooth(aes(x=rgyl2fc, y=rgl2fc), method=lm, se=FALSE, inherit.aes = FALSE) +xlab("Log2FoldChange Young vs Regenerated Young") +ylab("Log2FoldChange Old vs Regenerated Old") +scale_colour_manual(values=c("darkcyan", "orangered"))+theme(legend.position = "bottom", legend.title = element_blank())
```

That looks to be an emphatic no!

#Nikos Plot

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

#View(nmkcr)
gm <- subset(nmkcr, nmkcr$sig == TRUE)

ggplot(gm, aes(x=nl2fc, y=rgyl2fc)) +geom_point()


hist(log(gm$OR1b))
hist(log(gm$OR2b))

dgm <- data.frame(gm$OR1a, gm$OR2a, gm$OR2a, gm$OR2b, gm$p_value, gm$dir, gm$rgup)
dgm <- melt(dgm, id.vars=c("gm.p_value", "gm.dir", "gm.rgup"))

ggplot(dgm, aes(x=value, fill=gm.rgup)) +geom_histogram(binwidth=0.25) +scale_x_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8)) +geom_point(aes(x=value, y=0, colour=gm.rgup))

ggplot(dgm, aes(x=value)) +geom_density() +scale_x_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8)) +geom_point(aes(x=value, y=0, colour=gm.dir)) 


+facet_wrap(~gm.dir)

```


#What effect do the Housekeeping genes have?

Let's look at our data without normalising via the housekeeping genes. Is our aging effect stable without doing this normalisation? 



```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

bcm <- subset(bcun, rownames(bcun) %in% rownames(bc))

oldmean <- rowMeans(bcm[,c(2,11,12,13,14,23,24)])
youngmean <- rowMeans(bcm[,c(3,4,9,10,15,16,21,22)])
OvYl2fc <- foldchange2logratio(foldchange(youngmean, oldmean))
nsOvY <- OvYl2fc

plot(OvYl2fc, codeset$log2FoldChange, xlab="Nanostring l2fc", ylab="Transcriptome l2fc", main= "New data only, all old vs all young, normalised by int controls only", pch=16, cex=0.8, col=adjustcolor(alpha=0.5, "dark blue")) +
lines(OvYl2fc[(OvYl2fc >1)&(codeset$log2FoldChange >0)], codeset$log2FoldChange[(OvYl2fc >1)&(codeset$log2FoldChange >0)], lty=0, col="forest green", type="p", pch=16, cex=0.8) + 
lines(OvYl2fc[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], codeset$log2FoldChange[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], lty=0, col="forest green", type="p", pch=16, cex=0.8) + 
rect(1,1,10,10, col=adjustcolor(alpha=0.2, "forest green"), lty=0)+ rect(-10,-10,-1,-1, col=adjustcolor(alpha=0.2, "forest green"), lty=0) + text(2,1, labels=paste(length(OvYl2fc[(OvYl2fc >1)&(codeset$log2FoldChange >0)&!is.na(OvYl2fc)]), "genes"), col="forest green") +grid()+ 
text(-2,-1, labels=paste(length(OvYl2fc[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)&!is.na(OvYl2fc)]), "genes"), col="forest green") +
text(OvYl2fc[(OvYl2fc >1)&(codeset$log2FoldChange >0)], codeset$log2FoldChange[(OvYl2fc >1)&(codeset$log2FoldChange >0)], labels=codeset$shortname[(OvYl2fc >1)&(codeset$log2FoldChange >0)], col="dark green", offset=0.5,1.3, cex=0.65) +
text(OvYl2fc[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], codeset$log2FoldChange[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], labels=codeset$shortname[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], col="dark green", offset=0.5,1.3, cex=0.65)

```

Removing the normalisation doesn't seem to have had too much effect on the aging effect here. Let's look at the PCA. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
bcmin <- bcun[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 200 markers, normalised by internal controls only")+ scale_colour_manual(values=c("cornflower blue", "red3"))



```

We still return our old vs young separation, albeit split across two PCs. What about if we construct a PCA using just our housekeeping genes?

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
bcmin <- subset(bcun, !(rownames(bcun) %in% rownames(bc)))
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=TRUE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 5 housekeeping genes")+ scale_colour_manual(values=c("cornflower blue", "red3"))



```

And what about the age effect seen when comparing old and young usign our 61 significant markers? Is that stable when we don't normalise via the housekeeping genes?


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
bcmin <- subset(bcun, (rownames(bcun) %in% rownames(dbcg)))
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 61 markers (normalised by internal controls only)")+ scale_colour_manual(values=c("cornflower blue", "red3"))



```

We still return the effect: our aging signature isn't an artefact of normalisation. This is heartening. 
