---
title: "NanoStringFeb17"
author: "Luke Hayden"
date: "8th March 2017"
output: html_document
---
Second round of analyses on the February 2017 Nanostring dataset, combining old and new analyses. Here, I have done some more in-depth analyses on the variability of each marker (coefficients of variation and their effects on age signal), ranked markers by testing each for a significant old vs young difference, applied this test to improve old vs young signal, examined how regeneration affects these markers and examined the effect of normalisation on these results. 


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

setwd("~/nstringfeb17")

#source("https://bioconductor.org/biocLite.R")
#biocLite("DESeq2")
source("https://bioconductor.org/biocLite.R")
biocLite("DESeq2")

library(ggplot2)
library(DESeq2)
library(rmarkdown)
library(pracma)
library(gplots)
library(RColorBrewer)
library(reshape2)
library(scales)
library(ggbiplot)
library(gtools)
library(devtools)
library(NanoStringNorm)
library(ggthemes)
library(grid)
library(gridExtra)
library(sjstats)




multiplot <- function(..., plotlist = NULL, file, cols = 1, layout = NULL) {
  require(grid)
  
  plots <- c(list(...), plotlist)
  
  numPlots = length(plots)
  
  if (is.null(layout)) {
    layout <- matrix(seq(1, cols * ceiling(numPlots/cols)),
                     ncol = cols, nrow = ceiling(numPlots/cols))
  }
  
  if (numPlots == 1) {
    print(plots[[1]])
    
  } else {
    grid.newpage()
    pushViewport(viewport(layout = grid.layout(nrow(layout), ncol(layout))))
    
    for (i in 1:numPlots) {
      matchidx <- as.data.frame(which(layout == i, arr.ind = TRUE))
      
      print(plots[[i]], vp = viewport(layout.pos.row = matchidx$row,
                                      layout.pos.col = matchidx$col))
    }
  }
}
```

##Data import & Normalisation
Following samples:

Nikos's samples used for the RNAseq:

Old mixed sex pools (O1,2)

Young mixed sex pools (Y1,2)

Old regenerated mixed sex pools (OR1,2)

Young regenerated mixed sex pools (YR1,2)

Each sample was run twice separately: a & b replicates

My new samples

Female old pools (O5,O7,O8,O9)

Female young pools (Y5,Y6,Y7,Y8)


From the previous dataset (Oct 2016), we can also include:

4 individual large males (MOIA,B,C,D)

Male old pools (MOA,B)

Female old pools (FOA,B)

Male young pools (MYA,B)

Female young pools (FYA,B)


In the case of all of my samples, RNA was extracted from limbs T4-5. 
Normalisation is done via NanostringNorm, normalising via both internal controls and housekeeping genes. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}
codeset <- read.table("CodeSet.txt")
codeset$shortname <- substr(codeset$Customer_Identifier, 1,2)
Cht <- read.table("CodeSetCountdata.txt")

data.raw <- read.markup.RCC()
all.norm <- NanoStringNorm(x = data.raw, anno = NA, CodeCount ='sum', Background ="mean", SampleContent ='housekeeping.sum', round.values = FALSE, take.log = FALSE,return.matrix.of.endogenous.probes = TRUE);



colnames(all.norm) <- c("O1a", "O2a", "Y1a", "Y2a", "OR1a", "OR2a", "YR1a", "YR2a", "Y5a", "Y6a", "O5a", "O7a", "O1b", "O2b", "Y1b", "Y2b", "OR1b", "OR2b", "YR1b", "YR2b", "Y7", "Y8", "O8", "O9" )



#colnames(both) <- c("O1a", "O2a", "Y1a", "Y2a", "OR1a", "OR2a", "YR1a", "YR2a", "FY5a", "FY6a", "FO5a", "FO7a", "O1b", "O2b", "Y1b", "Y2b", "OR1b", "OR2b", "YR1b", "YR2b", "FY7", "FY8", "FO8", "FO9", "FYA", "FOA", "MYA", "MOA","FYB",	"FOB",	"MYB",	"MOB",	"MOIA",	"MOIB","MOIC",	"MOID")
```

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}


ctg1nhk <- read.xls.RCC(x= "RCCCollector_import1_7-2-17-nohk.xlsx", sheet=1)
ctg2nhk <- read.xls.RCC(x= "RCCCollector_import1_8-2-17-nohk.xlsx", sheet=1)

c1.unnorm <- NanoStringNorm(x = ctg1nhk, anno = NA, CodeCount ='sum', Background ="mean", SampleContent ='top.geo.mean', round.values = FALSE, take.log = FALSE,return.matrix.of.endogenous.probes = TRUE);
c2.unnorm <- NanoStringNorm(x = ctg2nhk, anno = NA, CodeCount ='sum', Background ="mean", SampleContent ='top.geo.mean', round.values = FALSE, take.log = FALSE,return.matrix.of.endogenous.probes = TRUE);


bcun <- cbind(as.data.frame(c1.unnorm), as.data.frame(c2.unnorm))


colnames(bcun) <- c("O1a", "O2a", "Y1a", "Y2a", "OR1a", "OR2a", "YR1a", "YR2a", "Y5a", "Y6a", "O5a", "O7a", "O1b", "O2b", "Y1b", "Y2b", "OR1b", "OR2b", "YR1b", "YR2b", "Y7", "Y8", "O8", "O9" )

bcung <- bcun[,c(2:6,9:24)]


```

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}
bc7 <- bc[,1:8]
bc8 <- bc[,13:20]
mcomp <- bc7
mcomp$gene <- paste(codeset$Customer_Identifier, codeset$DrosoTopHit) 
mcomp <- melt(mcomp, id.vars=c("gene"))
m8 <- melt(bc8)
mcomp$rep2 <- m8$value

#ggplot(mcomp, aes(x=value, y=rep2, color=variable)) +facet_wrap(~variable, ncol=2) +xlab("Nanostring Expression (Rep a)")+ylab("Nanostring Expression (Rep b)")+geom_point(size = 1.5) + scale_y_log10()+scale_x_log10()+scale_color_brewer(palette="Paired") +geom_smooth(method=lm)

```

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
nbc <- bc[,c(1:8,13:20)]
#plot(rowMeans(nbc), codeset$meanRPKM, log="xy", pch=16, col=adjustcolor("red3", alpha=0.5), #xlab="Mean Nanostring expression", ylab= "Mean transciptomic expression (RPKMs)") +grid()


```


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}
cscounts <- read.table("CodeSetCountdata.txt")

colnames(cscounts) <- colnames(nbc[,1:8])
csRPKMs <- cscounts
NbyL <- as.data.frame(codeset$Length %o% colSums(csRPKMs))
csRPKMs <- csRPKMs/NbyL
csRPKMs <- csRPKMs*1000000000

mnbc <- nbc
mnbc$gene <- paste(codeset$Customer_Identifier, codeset$DrosoTopHit) 
mnbc<- melt(mnbc, id.vars=c("gene"))
mcsr <- melt(csRPKMs)
mnbc$trexp <- mcsr$value
mnbc$sample <- gsub("[ab]", "", mnbc$variable)

#ggplot(mnbc, aes(x=value, y=trexp, color=sample)) +facet_wrap(~variable, ncol=4) +xlab("Nanostring Expression")+ylab("Transcriptomic Expression (RPKM)")+theme(panel.grid.minor.x=element_blank())+geom_point(size = 1) + scale_y_log10()+scale_x_log10()+scale_color_brewer(palette="Paired")+geom_smooth(method=lm)


```

#Old vs young logfoldchange in Nanostring vs Transcriptome
This is a basic measure of the quality of our markers

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

oldmean <- rowMeans(bc[,c(2,13,14)])
youngmean <- rowMeans(bc[,c(3,4,15,16)])
OvYl2fc <- foldchange2logratio(foldchange(youngmean, oldmean))
nikosOvY <- OvYl2fc

plot(OvYl2fc, codeset$log2FoldChange, xlab="Nanostring l2fc", ylab="Transcriptome l2fc", main= "New data only, all old vs young (Nikos samples only)", pch=16, cex=0.8, col=adjustcolor(alpha=0.5, "dark blue")) +
lines(OvYl2fc[(OvYl2fc >1)&(codeset$log2FoldChange >0)], codeset$log2FoldChange[(OvYl2fc >1)&(codeset$log2FoldChange >0)], lty=0, col="forest green", type="p", pch=16, cex=0.8) + 
lines(OvYl2fc[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], codeset$log2FoldChange[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], lty=0, col="forest green", type="p", pch=16, cex=0.8) + 
rect(1,1,10,10, col=adjustcolor(alpha=0.2, "forest green"), lty=0)+ rect(-10,-10,-1,-1, col=adjustcolor(alpha=0.2, "forest green"), lty=0) + text(2,1, labels=paste(length(OvYl2fc[(OvYl2fc >1)&(codeset$log2FoldChange >0)&!is.na(OvYl2fc)]), "genes"), col="forest green") +grid()+ 
text(-2,-1, labels=paste(length(OvYl2fc[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)&!is.na(OvYl2fc)]), "genes"), col="forest green") +
text(OvYl2fc[(OvYl2fc >1)&(codeset$log2FoldChange >0)], codeset$log2FoldChange[(OvYl2fc >1)&(codeset$log2FoldChange >0)], labels=codeset$shortname[(OvYl2fc >1)&(codeset$log2FoldChange >0)], col="dark green", offset=0.5,1.3, cex=0.65) +
text(OvYl2fc[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], codeset$log2FoldChange[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], labels=codeset$shortname[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], col="dark green", offset=0.5,1.3, cex=0.65)

```

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

oldmean <- rowMeans(bc[,c(2,11,12,13,14,23,24)])
youngmean <- rowMeans(bc[,c(3,4,9,10,15,16,21,22)])
OvYl2fc <- foldchange2logratio(foldchange(youngmean, oldmean))
nsOvY <- OvYl2fc

plot(OvYl2fc, codeset$log2FoldChange, xlab="Nanostring l2fc", ylab="Transcriptome l2fc", main= "New data only, all old vs all young", pch=16, cex=0.8, col=adjustcolor(alpha=0.5, "dark blue")) +
lines(OvYl2fc[(OvYl2fc >1)&(codeset$log2FoldChange >0)], codeset$log2FoldChange[(OvYl2fc >1)&(codeset$log2FoldChange >0)], lty=0, col="forest green", type="p", pch=16, cex=0.8) + 
lines(OvYl2fc[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], codeset$log2FoldChange[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], lty=0, col="forest green", type="p", pch=16, cex=0.8) + 
rect(1,1,10,10, col=adjustcolor(alpha=0.2, "forest green"), lty=0)+ rect(-10,-10,-1,-1, col=adjustcolor(alpha=0.2, "forest green"), lty=0) + text(2,1, labels=paste(length(OvYl2fc[(OvYl2fc >1)&(codeset$log2FoldChange >0)&!is.na(OvYl2fc)]), "genes"), col="forest green") +grid()+ 
text(-2,-1, labels=paste(length(OvYl2fc[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)&!is.na(OvYl2fc)]), "genes"), col="forest green") +
text(OvYl2fc[(OvYl2fc >1)&(codeset$log2FoldChange >0)], codeset$log2FoldChange[(OvYl2fc >1)&(codeset$log2FoldChange >0)], labels=codeset$shortname[(OvYl2fc >1)&(codeset$log2FoldChange >0)], col="dark green", offset=0.5,1.3, cex=0.65) +
text(OvYl2fc[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], codeset$log2FoldChange[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], labels=codeset$shortname[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], col="dark green", offset=0.5,1.3, cex=0.65)

```

This gives us 73 genes that follow the same direction in log2foldchange in both RNAseq and Nanostring data, while also having a reasonably strong l2fc in the latter (>1). 

#Variability of markers

##Residuals
Calculate residuals for linear regression between RNAseq and Nanostring expression data. Take out O1a, YR1a, YR2b. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

cmnbc <- subset(mnbc, mnbc$variable != "O1a")
cmnbc <- subset(cmnbc, cmnbc$variable != "YR1a")
cmnbc <- subset(cmnbc, cmnbc$variable != "YR2a")

res <- data.frame(resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "O2a"))))
res <- cbind(res, resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "Y1a"))), resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "Y2a"))), resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "OR1a"))), resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "OR2a"))), resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "O1b"))), resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "O2b"))), resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "Y1b"))), resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "Y2b"))), resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "OR1b"))), resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "OR2b"))), resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "YR1b"))), resid(lm(formula = value ~ trexp, data=subset(cmnbc, cmnbc$variable == "YR2b"))))

colnames(res) <- c("O2a", "Y1a", "Y2a", "OR1a", "OR2a", "O1b", "O2b", "Y1b", "Y2b", "OR1b", "OR2b", "YR1b", "YR2b")
row.names(res) <- paste(codeset$Customer_Identifier)

tres<-as.data.frame(t(abs(res)))
tres<- rbind(tres,colMeans(tres))
ttres <- as.data.frame(t(tres))
ttres$marker <- substr(rownames(ttres),1,2)

colnames(ttres) <- c("O2a", "Y1a", "Y2a", "OR1a", "OR2a", "O1b", "O2b", "Y1b", "Y2b", "OR1b", "OR2b", "YR1b", "YR2b", "mean", "marker")

mtres <- melt(as.data.frame(ttres), id.vars=c("mean", "marker"))

mtres <- mtres[order(mtres$mean),]
l <- ttres$marker[order(ttres$mean)]

mtres$meanfac <- as.factor(mtres$mean)
#mtres$marker <- as.factor(mtres$marker,levels=mtres$marker)


ggplot(mtres, aes(x=meanfac, y=value))+xlab("Nanostring Marker")+ylab("Residuals (absolute)")+geom_point(shape=95) + theme(axis.text.x = element_text(angle = 80, hjust = 1, size=5)) +geom_point(aes(x=meanfac, y=mean), geom="point", colour="darkcyan")+ylim(0,150) +scale_x_discrete(labels=l)



#plot(rowMeans(abs(res)), abs(OvYl2fc), xlim=c(0,150), pch=16, col=adjustcolor("darkblue", alpha=0.4), xlab=)


#plot(rowMeans(res), OvYl2fc, xlim=c(0,150), pch=16, col=adjustcolor("darkblue", alpha=0.4), xlab="Mean absolute residual value")
```


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
oldmean <- rowMeans(bc[,c(2,11,12,13,14,23,24)])
youngmean <- rowMeans(bc[,c(3,4,9,10,15,16,21,22)])
OvYl2fc <- foldchange2logratio(foldchange(youngmean, oldmean))

rv <- data.frame(abs(rowMeans(res)))
rv <- cbind(rv, OvYl2fc, codeset$log2FoldChange)
colnames(rv) <- c("meanabsres", "NSOvYl2fc", "TransOvYl2fc") 
rv$rightdir <- ((OvYl2fc >1)&(codeset$log2FoldChange >0) |(OvYl2fc <(-1))&(codeset$log2FoldChange <0))
rv$absNS <- abs(rv$NSOvYl2fc)

```

If the failure of much of the markers to replicate the old vs young expression difference is due to inaccuracy in measurement of expression, those genes with the lowest residuals will have highest old vs young expression difference. Samples in green are our 73 markers that follow the "right direction" in log2foldchange between old and young. 



```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
rv$nikosOvY <- nikosOvY
rv$absnik <- abs(nikosOvY)

#ggplot(rv, aes(x=nikosOvY, y=meanabsres, colour=rightdir)) +geom_point(size=1) +scale_color_manual(values=c("dark blue", "forest green"),name="Old vs young L2FC",breaks=c(TRUE, FALSE), labels=c("OvY l2fc >1, <-1, expected direction", "Other"))+theme(legend.position = "top") +xlab("Old versus Young Log2FoldChange (Nanostring, Nikos' data only)") + ylab("Mean absolute residual value")

ggplot(rv, aes(x=absnik, y=meanabsres, colour=rightdir)) +geom_point(size=1) +scale_color_manual(values=c("dark blue", "forest green"),name="Old vs young L2FC",breaks=c(TRUE, FALSE), labels=c("OvY l2fc >1, <-1, expected direction", "Other"))+theme(legend.position = "top") +xlab("Absolute Old versus Young Log2FoldChange (Nanostring, Nikos' data only)") + ylab("Mean absolute residual value")+ stat_smooth(aes(x=absNS, y=meanabsres), method=lm, se=FALSE, inherit.aes=FALSE)

```


There is no correlation between residual value and l2fc: this means that residual variation isn't driving logfoldchange in epxression between old and young. 

##Coefficients of Variation

We want the coefficient of variation for each gene; this is unitless and best allows us to compare between genes. Here we compute the coefficient of variation for a linear model of Nanostring expression vs RNAseq expression produced for each marker. Where this value is high, the former is a poor predictor of the latter. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
gnbc <- nbc[,c(2:6,9:16)]
gt <- cbind(cscounts, cscounts)[,c(2:6,9:16)]

nst <- cbind(gnbc,gt)
cvvec <- c()

for (i in 1:length(rownames(nst))){
  cvvec <- c(cvvec, cv(lm(formula = ns~t, data=data.frame(ns=melt(nst[i,1:13])$value, t=melt(nst[i,14:26])$value))))
}

hist(cvvec, col="forest green", xlab="Coefficient of variation", main="Distribution of Coefficients of Variation")

rv$cv <- cvvec
rv$nameshort <- substr(rownames(rv), 1,2)
rv$crn <- as.factor(rv$cv)

ggplot(rv, aes(x=crn, y=cv)) +geom_point(colour="darkcyan") +scale_x_discrete(labels=rv$nameshort[order(rv$cv)]) + theme(axis.text.x = element_text(angle = 80, hjust = 1, size=5)) +xlab("Marker ID") + ylab("Coefficient of variation (Nanostring vs RNAseq") +ggtitle("Distribution of Coefficients of variation") +scale_colour_manual(values=c("darkcyan"))

ggplot(rv, aes(x=cv, y=absnik,colour=rightdir))+geom_point(size=1)+scale_color_manual(values=c("dark blue", "forest green"),name="Old vs young L2FC",breaks=c(TRUE, FALSE), labels=c("OvY l2fc >1, <-1, expected direction", "Other"))+theme(legend.position = "top") + stat_smooth(aes(x=cv, y=absnik), method=lm, se=FALSE, inherit.aes=FALSE) +ylab("Absolute Old versus Young Log2FoldChange (Nanostring, Nikos' data only)") + xlab("Coefficient of Variation") +ggtitle("Does Coefficient of Variation predict which genes show age-related change?")

```

This indicates that coefficient of determination doesn't determine whether or not a marker shows an aging signature. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

ggplot(rv, aes(x=cv, y=meanabsres,colour=rightdir))+geom_point(size=1)+scale_color_manual(values=c("dark blue", "forest green"),name="Old vs young L2FC",breaks=c(TRUE, FALSE), labels=c("OvY l2fc >1, <-1, expected direction", "Other"))+theme(legend.position = "top") + stat_smooth(aes(x=cv, y=meanabsres), method=lm, se=FALSE, inherit.aes=FALSE) +  scale_y_continuous(trans='log2')+ylab("Mean absolute residuals") + xlab("Coefficient of Variation") +ggtitle("How do residuals relate to coefficients of variation?")

```
  
  
```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
rv$meanexpns <- rowMeans(gnbc)
rv$meanexpt <- rowMeans(cscounts)



ggplot(rv, aes(x=cv, y=meanexpns)) +geom_point()+scale_color_manual(values=c("dark blue"))+ stat_smooth(se=FALSE,method=lm) +ylab("Mean Nanostring expression level") +xlab("Coefficient of variation Nanostring vs Transcriptome") +ggtitle("Is expression level correlated with coefficient of variation?")

#+geom_point(aes(x=cv, y=meanexpt), colour="forest green", alpha=0.5) +ylim(0,800)
```
  
So higher expression levels are correlated with lower coefficients of variation.   
  
Let's also look at the coefficients of variation when obtained when we compare Nanostring replicates (we have two good Nanostring replicates for O2, Y1, Y2, OR1, OR2). 
  
```{r, echo=FALSE, warning=FALSE, message=FALSE}
rep1 <- nbc[,c(2:6)]
rep2 <- nbc[,c(10:14)]



rc <- cbind(nbc[,c(10:14)], nbc[,c(10:14)])
cvrep <- c()

for (i in 1:length(rownames(rc))){
  cvrep <- c(cvrep, cv(lm(formula = ns~t, data=data.frame(ns=melt(nst[i,1:5])$value, t=melt(nst[i,6:10])$value))))
}

#hist(cvvec, col="cornflower blue", xlab="Coefficient of variation") +hist(cvrep, col=adjustcolor(alpha=0.5, "red3"), add=TRUE)

rv$cvrep <- cvrep

mrv <- melt(rv, id.vars=c("meanabsres","NSOvYl2fc" ,"TransOvYl2fc", "rightdir" ,"absNS","nikosOvY","absnik","nameshort","meanexpns","meanexpt", "crn"))

ggplot(mrv, aes(x=value, fill=variable))+geom_histogram(binwidth=0.2,alpha=0.5)+scale_fill_manual(name="x", values=c("darkcyan", "orangered"),breaks=c("cv", "cvrep"), labels=c("Nanostring vs RNAseq", "Between Nanostring replicates"))+theme(legend.position = "bottom", legend.title = element_blank()) +xlab("Coefficient of variation value")+ggtitle("Comparing coefficients of variation")

ggplot(rv, aes(x=cv, y=cvrep,colour=rightdir))+geom_point(size=1) + stat_smooth(aes(x=cv, y=cvrep), method=lm, se=FALSE, inherit.aes=FALSE) +ylab("Coefficient of variation comparing Nanostring replicates") + xlab("Coefficient of Variation NS vs RNAseq") +ggtitle("Comparing coefficients of variation") +scale_color_manual(values=c("dark blue", "forest green"),name="Old vs young L2FC",breaks=c(TRUE, FALSE), labels=c("OvY l2fc >1, <-1, expected direction", "Other"))+theme(legend.position = "top")

```
  
Our coefficients of variation are higher when we compare Nanostring vs RNAseq, as we would expect. However, these two correlate reasonably well with one another; the genes that vary Nanostring vs RNAseq also tend to vary between replicates.   
  
  
#PCA: new samples only
Remove the low quality samples (O1a, YR1a, YR2a). NOw lets go back to the PCAs and see how we can best optimise our aging signal. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
bcw <-bc[,c(2:6,9:24)]
tbc <- t(bcw)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "OR", "OR", "Y", "Y", "O", "O","O", "O", "Y", "Y", "OR", "OR", "YR", "YR", "Y", "Y", "O", "O")

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'top') +ggtitle("PCA of all samples (normalised via housekeeping genes and internal controls NanoStringNorm")+ scale_colour_manual(values=c("cornflower blue", "forest green" ,"red3", "black"))

p<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'top') +ggtitle("PCA of all samples (normalised via housekeeping genes and internal controls NanoStringNorm")+ scale_colour_manual(values=c("cornflower blue", "forest green" ,"red3", "black"))


ggsave(plot=p,height=10,width=10,dpi=200, filename=paste("3PCAnewsamples.pdf"), useDingbats=FALSE, limitsize = FALSE)



```     
   
Now that the bad samples have been removed, we can now see separation along PC2 between old and young. 
     
  
```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
tbc <- t(both[,c(2:6,9:36) ])
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "OR", "OR", "FY", "FY", "FO", "FO","O", "O", "Y", "Y", "OR", "OR", "YR", "YR", "FY", "FY", "FO", "FO", "FY", "FO", "MY", "MO","FY", "FO", "MY", "MO", "MO", "MO", "MO", "MO")

#ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'top') +ggtitle("PCA of samples from both datasets")+ scale_colour_manual(values=c("cornflower blue", "red3", "dark blue","orangered", "darkcyan", "forest green", "coral", "black"))


```     
     
#PCA: find age signature
First, we'll use log2foldchange between old and young as our criterion. We can calculate this in two ways: using just Nikos old vs young or including our new old and young samples. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
mk <- data.frame(names(nikosOvY), codeset$log2FoldChange,nikosOvY, nsOvY)
colnames(mk) <- c("marker", "csl2fc", "nikl2fc", "nl2fc")
mk$smdir <- mk$csl2fc * mk$nikl2fc >0
mk$nsmdir <- mk$csl2fc * mk$nl2fc >0


par(mfrow=c(2,1))
hist(abs(mk$nikl2fc[mk$smdir==TRUE]), xlab="Absolute Log2FoldChange Old vs Young, Nikos samples only", main=paste(length(mk$nikl2fc[mk$smdir==TRUE]), "Markers with same l2fc direction RNAseq and Nanostring"), col=adjustcolor("cornflower blue", alpha=0.7)) 
hist(abs(mk$nl2fc[mk$smdir==TRUE]), xlab="Absolute Log2FoldChange Old vs Young, Nikos + new samples", main=paste(length(mk$nikl2fc[mk$smdir==TRUE]), "Markers with same l2fc direction RNAseq and Nanostring"), col=adjustcolor("red3", alpha=0.7)) 




```

Construct a series of PCAs, using progressively more stringent cut-off points in relation to old vs young log2foldchange in Nikos' samples.  


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
bcmin <- bc[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pa<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 195 markers")+ scale_colour_manual(values=c("cornflower blue", "red3"))


bcmin <- subset(bc, mk$smdir ==TRUE)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pb<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y l2fc dir same: 180 markers")+ scale_colour_manual(values=c("cornflower blue", "red3"))

bcmin <- subset(bc, abs(mk$nikl2fc) >0.5 & mk$smdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pc<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y, l2fc>1,<-1: ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))

bcmin <- subset(bc, abs(mk$nikl2fc) >1 & mk$smdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pd<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y, l2fc>1,<-1: ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))

bcmin <- subset(bc, abs(mk$nikl2fc) >2 & mk$smdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pe<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y, l2fc>2,<-2: ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))

bcmin <- subset(bc, abs(mk$nikl2fc) >3 & mk$smdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pf<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y, l2fc>3,<-3: ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))



multiplot(pa, pb, pc, pd,pe,pf, cols=3)

```

And now if we use l2fc in all old vs young of our new samples as the criterion, we should expect to see somewhat better separation. However, this is somewhat circular as we are choosing genes on the basis of them differing in expression between the two groups

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
bcmin <- bc[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pa<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 195 markers")+ scale_colour_manual(values=c("cornflower blue", "red3"))


bcmin <- subset(bc, mk$nsmdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

  
pb<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y l2fc dir same: ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))
                                                                                                                                                                                            bcmin <- subset(bc, abs(mk$nl2fc) >0.5 & mk$nsmdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pc<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y, l2fc>1,<-1: ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))

bcmin <- subset(bc, abs(mk$nl2fc) >1 & mk$nsmdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pd<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y, l2fc>1,<-1: ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))

bcmin <- subset(bc, abs(mk$nl2fc) >2 & mk$nsmdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pe<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y, l2fc>2,<-2: ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))

bcmin <- subset(bc, abs(mk$nl2fc) >3 & mk$nsmdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pf<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y, l2fc>3,<-3: ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))



multiplot(pa, pb, pc, pd,pe,pf, cols=3)
```



#Age signal content of markers
We have 195 markers, which vary in terms of how well they separate old vs young. Let's rank them in terms of their old vs young log2foldchange and look at how consistent each of them is. 


##Examining Individual markers by their Old vs Young Difference


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
mkc <- bc[,c(2,3,4,9,10,11,12,13:16,21:24)]
mkc$mean <- rowMeans(mkc)

mkc <- cbind(mkc, mk)

mkc$name <- codeset$shortname
mkc$db <- codeset$DrosoTopHit
mkc$dir <- ""
mkc$dir[rv$NSOvYl2fc >0] <- "downregulated with age"
mkc$dir[rv$NSOvYl2fc <0] <- "upregulated with age"

mkc$name <- factor(mkc$name, levels=mkc$name[order(mkc$nl2fc)])
mkc$gr <- abs(mk$nl2fc) >1 & mk$nsmdir ==TRUE

mkcoy <- cbind(mkc[,c(1,6:9,14:15)], mkc[,c(2:5,10:13)])

t.result <- apply(mkcoy, 1, function (x) t.test(x[1:7],x[8:15]))
mkc$p_value <- unlist(lapply(t.result, function(x) x$p.value))
mkc$t_stat <- unlist(lapply(t.result, function(x) x$statistic))

mmkc <- melt(mkc, id.vars=c("mean", "marker", "csl2fc", "nikl2fc", "nl2fc", "smdir", "nsmdir","name", "db", "dir", "gr", "p_value", "t_stat"))
mmkc$age <- substr(mmkc$variable, 1,1)
mmkc$type <- "New samples"
mmkc$type[grepl("1|2", mmkc$variable)] <- "Nikos samples" 

p<-ggplot(mmkc, aes(x=age, y=value, color=age)) +facet_wrap(~name, nrow=1, scales="free_y")  +geom_boxplot(alpha=0.6, aes(fill=gr,x=age, y=value))+scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young"))+ylab("Nanostring expression") +xlab("Age")+ scale_fill_manual(values=c("white", "azure4"))+ theme(axis.text.y = element_text(angle = 90, hjust =0.5, size=7))+ theme(legend.direction = 'horizontal', legend.position = 'top') +geom_point(aes(x=age, y=value, shape=type)) 

ggsave(plot=p,height=10,width=100,dpi=200, filename=paste("Ageeffectby marker1.pdf"), useDingbats=FALSE, limitsize = FALSE)

```

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
mmkc$ud <- mmkc$value/mmkc$mean
mmkc$nl <- paste(substr(mmkc$nl2fc,1,6), mmkc$name)
mmkcup <- subset(mmkc, mmkc$dir == "upregulated with age")

mmkcdown <- subset(mmkc, mmkc$dir == "downregulated with age")

ggplot(mmkcup, aes(x=nl, y=ud, color=age,)) +geom_point(aes(shape=type), alpha=0.6) +xlab("Marker (ordered by OvY log2FoldChange")+ theme(axis.text.x = element_text(angle = 60, hjust = 1)) +scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young")) +ylab("Nanostring expression (as proportion of mean)")+  scale_y_continuous(trans='log2', breaks=c(0,0.25,0.5,1,2,4))+geom_hline(yintercept=1, colour = "black", linetype=2)  +ggtitle("Expression of genes upregulated with age") +stat_summary(fun.y=mean, geom="point", shape=3, size=3)+scale_x_discrete(labels=subset(mkc, mkc$dir == "upregulated with age")$name[order(subset(mkc, mkc$dir == "upregulated with age")$nl2fc)])

ggplot(mmkcdown, aes(x=nl, y=ud, color=age,)) +geom_point(aes(shape=type), alpha=0.6) +xlab("Marker (ordered by OvY log2FoldChange")+ theme(axis.text.x = element_text(angle = 60, hjust = 1, size=5)) +scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young"))+ylab("Nanostring expression (as proportion of mean)")+  scale_y_continuous(trans='log2', breaks=c(0,0.25,0.5,1,2,4))+geom_hline(yintercept=1, colour = "black", linetype=2)  +ggtitle("Expression of genes downregulated with age")+stat_summary(fun.y=mean, geom="point", shape=3, size=3)+scale_x_discrete(labels=subset(mkc, mkc$dir == "downregulated with age")$name[order(subset(mkc, mkc$dir == "downregulated with age")$nl2fc)])



```

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
ggplot(mmkcup, aes(x=nl, y=ud, color=age,)) +geom_point(aes(shape=type), alpha=0.6) +xlab("Marker (ordered by OvY log2FoldChange")+ theme(axis.text.x = element_text(angle = 60, hjust = 1)) +scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young"))+ylab("Nanostring expression (as proportion of mean)")+  scale_y_continuous(trans='log2', breaks=c(0,0.25,0.5,1,2,4))+geom_hline(yintercept=1, colour = "black", linetype=2)  +ggtitle("Expression of genes upregulated with age") +geom_boxplot()+scale_x_discrete(labels=subset(mkc, mkc$dir == "upregulated with age")$name[order(subset(mkc, mkc$dir == "upregulated with age")$nl2fc)])

ggplot(mmkcdown, aes(x=nl, y=ud, color=age,)) +geom_point(aes(shape=type), alpha=0.6) +xlab("Marker (ordered by OvY log2FoldChange")+ theme(axis.text.x = element_text(angle = 60, hjust = 1, size=5)) +scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young"))+ylab("Nanostring expression (as proportion of mean)")+  scale_y_continuous(trans='log2', breaks=c(0,0.25,0.5,1,2,4)) +geom_hline(yintercept=1, colour = "black", linetype=2) +ggtitle("Expression of genes downregulated with age")+geom_boxplot()+scale_x_discrete(labels=subset(mkc, mkc$dir == "downregulated with age")$name[order(subset(mkc, mkc$dir == "downregulated with age")$nl2fc)])




```


##Ranking these markers by their consistency
We want markers that are consistent moreso than those that have large magnitude changes. How do we find the most consistent markers? Now we perform a t-test for each gene, looking for whether each gene has a significant difference in expression level between old and young samples. These are the p-values we get:

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}
mkc$sig <- mkc$p_value <0.05
sig <- mkc$p_value <0.05

hist(mkc$p_value, breaks=seq(from=0, to=1, by=0.05), col="forest green", main="P-values of old vs young differences by gene", xlab="p-value")

#plot(mkc$nl2fc, mkc$p_value, pch=16, col="darkblue", xlab="Old vs Young Log2Foldchange", ylab="Old vs Young t-test p-value") +points(mkc$nl2fc[mkc$p_value <0.05], -mkc$p_value[mkc$p_value <0.05], pch=16, col="forest green")

ggplot(mkc, aes(x=nl2fc, y=p_value, color=sig)) +geom_point() +scale_colour_manual(values=c("dark blue", "forest green"), breaks=c(TRUE, FALSE), labels=c("p<0.05", "p>0.05"), name="OvY t-test result") +xlab("Old vs Young log2foldchange") +ylab("Old vs Young t-test p-value")
```



```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}

mkc$name <- factor(mkc$name, levels=mkc$name[order(mkc$p_value)])

mmkc <- melt(mkc, id.vars=c("mean", "marker", "csl2fc", "nikl2fc", "nl2fc", "smdir", "nsmdir","name", "db", "dir", "gr", "p_value", "t_stat", "sig"))
mmkc$age <- substr(mmkc$variable, 1,1)
mmkc$type <- "New samples"
mmkc$type[grepl("1|2", mmkc$variable)] <- "Nikos samples" 


p<-ggplot(mmkc, aes(x=age, y=value, color=age)) +facet_wrap(~name, nrow=1, scales="free_y")  +geom_boxplot(alpha=0.6, aes(fill=sig,x=age, y=value))+scale_color_manual(values=c("cornflower blue", "red3"))+ylab("Nanostring expression") +xlab("Age")+ scale_fill_manual(values=c("white", "azure4"))+ theme(axis.text.y = element_text(angle = 90, hjust =0.5, size=7))+ theme(legend.direction = 'horizontal', legend.position = 'top') +geom_point(aes(x=age, y=value, shape=type)) 

ggsave(plot=p,height=10,width=100,dpi=200, filename=paste("Ageeffectby marker3.pdf"), useDingbats=FALSE, limitsize = FALSE)

```

Then, we can order our markers in relation to how significant (or not) the result we got was. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
mmkc$ud <- mmkc$value/mmkc$mean
mmkc$nl <- paste(substr(mmkc$p_value,1,4), mmkc$name)
mmkcup <- subset(mmkc, mmkc$dir == "upregulated with age")

mmkcdown <- subset(mmkc, mmkc$dir == "downregulated with age")

ggplot(mmkcup, aes(x=nl, y=ud, color=age)) +geom_point(aes(shape=type), alpha=0.6) +xlab("Marker (ordered by increasing OvY t-test p-value)")+ theme(axis.text.x = element_text(angle = 60, hjust = 1)) +scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young")) +ylab("Nanostring expression (as proportion of mean)")+  scale_y_continuous(trans='log2', breaks=c(0,0.25,0.5,1,2,4))+geom_hline(yintercept=1, colour = "black", linetype=2)  +ggtitle("Expression of genes upregulated with age") +stat_summary(fun.y=mean, geom="point", shape=3, size=3)+scale_x_discrete(labels=subset(mkc, mkc$dir == "upregulated with age")$name[order(subset(mkc, mkc$dir == "upregulated with age")$p_value)])



#subset(mkc, mkc$dir == "downregulated with age")$name[order(subset(mkc, mkc$dir == "downregulated with age")$nl2fc)]

ggplot(mmkcdown, aes(x=nl, y=ud, color=age)) +geom_point(aes(shape=type), alpha=0.6) +xlab("Marker (ordered by increasing OvY t-test p-value)")+ theme(axis.text.x = element_text(angle = 60, hjust = 1, size=5)) +scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young"))+ylab("Nanostring expression (as proportion of mean)")+  scale_y_continuous(trans='log2', breaks=c(0,0.25,0.5,1,2,4))+geom_hline(yintercept=1, colour = "black", linetype=2)  +ggtitle("Expression of genes downregulated with age")+stat_summary(fun.y=mean, geom="point", shape=3, size=3) +scale_x_discrete(labels=subset(mkc, mkc$dir == "downregulated with age")$name[order(subset(mkc, mkc$dir == "downregulated with age")$p_value)])


```


##Applying progressively more stringent cut-off points for marker choice on the basis of p-value

Hopefully, this should improve our old vs young separation. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
mk$p_value <- mkc$p_value
mk$sig <- sig


bcmin <- bc[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pa<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 195 markers")+ scale_colour_manual(values=c("cornflower blue", "red3"))


bcmin <- subset(bc, mk$nsmdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

  
pb<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y l2fc dir same: ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))
                                                                                                                                                                                            
bcmin <- subset(bc, mk$p_value <0.2 & mk$nsmdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pc<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y, p-value<0.2 ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))

bcmin <- subset(bc, mk$p_value <0.1 & mk$nsmdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pd<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y, p-value<0.1 ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))

bcmin <- subset(bc, mk$p_value <0.05 & mk$nsmdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pe<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y, , p-value<0.05 ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))

bcmin <- subset(bc, mk$p_value <0.02 & mk$nsmdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

pf<-ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle(paste("PCA of O vs Y, , p-value<0.02 ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))



multiplot(pa, pb, pc, pd,pe,pf, cols=3)
```


This one looks good:

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}
bcmin <- subset(bc, mk$p_value <0.05 & mk$nsmdir ==TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")



ggbiplot(bc.pca, choices=1:2, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'top')+ggtitle(paste("PCA of O vs Y, , p-value<0.05, ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))


ggbiplot(bc.pca, choices=3:4, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'top')+ggtitle(paste("PCA of O vs Y, , p-value<0.05 ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))
```

#t-test vs Log2FoldChange as criteria for marker choice

Are the 61 markers chosen above on the basis of their t-test p-values all amongst the 73 chosen earlier on the basis of their log2foldchange?

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}
pv <- rownames(subset(mkc, mkc$sig))
pl <- rownames(subset(mkc, mkc$gr))

venn(list("p-value for OvY t-test < 0.05"= rownames(subset(mkc, mkc$sig)), "OvY log2foldchange >1, <-1, direction of change as expected" = rownames(subset(mkc, mkc$gr))))



```

Not always!
To what degree do log2foldchange and t-test p-value correlate?

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}
mkc$absnl2fc <- abs(mkc$nl2fc)
ggplot(mkc, aes(y=p_value, x=nl2fc, colour=sig, fill=gr)) +geom_point(shape=21)  +scale_colour_manual(values=c("orangered", "forest green"), breaks=c(TRUE, FALSE), labels=c("p<0.05", "p>0.05"), name="t-test result") +scale_fill_manual(values=c("white", "black"),name="Old vs young L2FC",breaks=c(TRUE, FALSE), labels=c("OvY l2fc >1, <-1, expected direction", "Other")) +ylab("Old vs Young t-test p-value") +xlab("Old vs Young Log2FoldChange") 

ggplot(mkc, aes(y=p_value, x=absnl2fc, colour=sig, fill=gr)) +geom_point(shape=21)  +scale_colour_manual(values=c("orangered", "forest green"), breaks=c(TRUE, FALSE), labels=c("p<0.05", "p>0.05"), name="t-test result") +scale_fill_manual(values=c("black", "white"),name="Old vs young L2FC", labels=c("OvY l2fc >1, <-1, expected direction", "Other"))+ylab("Old vs Young t-test p-value") +xlab("Absolute Old vs Young Log2FoldChange") +stat_smooth(aes(y=p_value, x=absnl2fc), method=lm, se=FALSE, inherit.aes=FALSE) +ylim(0,1) 

```

What if we were to further refine our marker search by only taking those which are in both groups? These are the points with the dark centres and the green outlines above; those with a strong age-associated change that is also consistent. 

Let's try a PCA using just those markers that are in both lists:

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
bcmin <- subset(bc, mk$p_value <0.05 & mk$nsmdir ==TRUE & mkc$gr == TRUE)
l <- length(bcmin$O2a)
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")



ggbiplot(bc.pca, choices=1:2, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'top')+ggtitle(paste("PCA of O vs Y, , p-value<0.05, l2fc >1,<-1 ", toString(l), " markers"))+ scale_colour_manual(values=c("cornflower blue", "red3"))



```

This doesn't look terribly different from when we use all 61 markers chosen on the basis of our t-test results. 


#Expression levels of chosen markers as age signal?
```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}
#dbcg <- subset(bc, rv$NSOvYl2fc >0 &rv$rightdir ==TRUE)


dbcg <- bc[,c(2:4,9:16,21:24)]
dbcg$dir <- ""
dbcg$dir[rv$NSOvYl2fc >0] <- "downregulated with age"
dbcg$dir[rv$NSOvYl2fc <0] <- "upregulated with age"


dbcg <- cbind(dbcg, mkc[,c(27:29)])



dbcg <- subset(dbcg,  dbcg$sig==TRUE)


dbcg <- dbcg[,order(colnames(dbcg))]
mdbcg <- melt(dbcg, id.vars=c("dir", "p_value", "t_stat", "sig"))
mdbcg$age <- substr(mdbcg$variable,1,1)

#ggplot(mdbcg, aes(x=age, y=value, color=age)) +geom_boxplot() +ylim(0,80) +ggtitle("61 chosen markers: p<0.05")+scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young"))+ facet_wrap(~dir) +ylab("Nanostring expression (absolute counts across markers)") +xlab("Sample")
 
#ggplot(mdbcg, aes(x=variable, y=value, color=age)) +geom_boxplot() +ggtitle("61 chosen markers: p<0.05")+scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young"))+ylim(0,80)+ facet_wrap(~dir, ncol=1) +ylab("Nanostring expression (absolute counts across markers)") +xlab("Sample")+ theme(axis.text.x = element_text(angle = 60, hjust = 1)) +geom_point(shape=95, size=2)

```


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}
#print("Genes downregulated with age:")
#t.test(mdbcg$value[mdbcg$age=="O" &mdbcg$dir=="downregulated with age"], mdbcg$value[mdbcg$age=="Y"&mdbcg$dir=="downregulated with age"])

#print("Genes upregulated with age:")
#t.test(mdbcg$value[mdbcg$age=="O" &mdbcg$dir=="upregulated with age"], mdbcg$value[mdbcg$age=="Y"&mdbcg$dir=="upregulated with age"])



```

Lets normalise each gene's expression by its mean across these conditions. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

ndbcg <- as.matrix(cbind(dbcg[,c(2:8)], dbcg[,c(12:19)]) )
ndbcg <- as.data.frame(ndbcg/rowMeans(as.matrix(ndbcg)))
                       
ndbcg$dir <- dbcg$dir
ndbcg$marker <- rownames(ndbcg)
mndbcg <- melt(ndbcg, id.vars=c("dir", "marker"))
mndbcg$age <- substr(mndbcg$variable,1,1)

ggplot(mndbcg, aes(x=age, y=value, color=age)) +geom_boxplot() +ggtitle("61 chosen markers, normalised by mean expression")+scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young")) + facet_wrap(~dir) +ylab("Nanostring expression (normalised)") +xlab("Sample")+geom_hline(yintercept = 1, linetype=2)



mndbcg <- mndbcg[order(mndbcg$age),]

ggplot(mndbcg, aes(x=variable, y=value, color=age)) +geom_boxplot() +ggtitle("61 chosen markers, normalised by mean expression")+scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young"))+ facet_wrap(~dir, ncol=1)+ylab("Nanostring expression (normalised)") +xlab("Sample")+ theme(axis.text.x = element_text(angle = 60, hjust = 1)) +geom_hline(yintercept = 1, linetype=2) +geom_point(shape=95, size=3)


```

Now we see the signal more clearly.

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE, results="hide"}
print("Genes downregulated with age:")
t.test(mndbcg$value[mdbcg$age=="O" &mndbcg$dir=="downregulated with age"], mndbcg$value[mndbcg$age=="Y"&mndbcg$dir=="downregulated with age"])

print("Genes upregulated with age:")
t.test(mndbcg$value[mndbcg$age=="O" &mndbcg$dir=="upregulated with age"], mndbcg$value[mndbcg$age=="Y"&mndbcg$dir=="upregulated with age"])



```

#Age effect by marker
Still using the same samples: old vs young from the recent dataset only. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

mndbcg$type <- paste(mndbcg$age, " , ", mndbcg$dir)
mndbcg <- mndbcg[order(mndbcg$dir),]
mndbcg$mkty <- paste(substr(mndbcg$dir, 1, 2)," ", substr(mndbcg$marker, 1, 2))


ggplot(mndbcg, aes(x=mkty, y=value, color=age, fill=dir)) +geom_point(shape=95, size=4.5) +ggtitle("61 chosen markers, normalised by mean expression")+scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young"))+ylab("Nanostring expression (normalised)") +xlab("Sample")+ theme(axis.text.x = element_text(angle = 60, hjust = 1))+stat_summary(fun.y=mean, geom="point", shape=21)+ scale_fill_manual(values=c("white", "azure4"), name="Direction")+geom_hline(yintercept = 1, linetype=2) +geom_vline(xintercept="moo") + scale_y_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8) ) +scale_x_discrete(labels=substr(ndbcg$marker,1,2))


ggplot(mndbcg, aes(x=mkty, y=value, color=age)) +geom_boxplot(aes(fill=dir)) +ggtitle("61 chosen markers, normalised by mean expression")+scale_color_manual(values=c("cornflower blue", "red3"), name= "Age", labels=c("Old", "Young"))+ylab("Nanostring expression (normalised)") +xlab("Sample")+ theme(axis.text.x = element_text(angle = 60, hjust = 1))+ scale_fill_manual(values=c("white", "azure4"), name="Direction")+geom_hline(yintercept = 1, linetype=2) +geom_vline(xintercept="moo")+ scale_y_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8) )+scale_x_discrete(labels=substr(ndbcg$marker,1,2))



```


Each marker is not 100% predictive, some old samples mix among the young ones. 


#Coefficient of variation vs Old vs Young t-test


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
mkc$cv <- cvvec

ggplot(mkc, aes(y=p_value, x=cv, colour=sig)) +geom_point() +stat_smooth(aes(y=p_value, x=cv), method=lm, se=FALSE, inherit.aes=FALSE) +ylim(0,1) +scale_colour_manual(labels=c("p<0.05", "p>0.05"), name="t-test result", values=c("black", "forest green")) +ggtitle("Does Coefficient of Variation predict old vs young change?") +xlab("Coefficient of Variation (Nanostring vs RNAseq)") +ylab("Old vs Young t-test p-value")



```

#Marker Identity
Drosophila top BLAST hits:
```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
redcs <- subset(codeset, sig == TRUE)
print(redcs$DrosoTopHit[order(redcs$DrosoTopHit)])
#print(redcs$TopHit)

```

#Project regenerated samples onto PCA of Old vs Young

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
bcg <- subset(bc, sig ==TRUE)
bcg <- bcg[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcg)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
rd.pca <- bc.pca
rd<- bcg
rg <-subset(bc, sig ==TRUE)
rg <- rg[,c(5:6,17:20)]


rg.sc <- scale(t(rg), center= rd.pca$center)
rg.pred <- rg.sc %*% rd.pca$rotation 


rgd.pca <- rd.pca
rgd.pca$x <- rbind(rd.pca$x, rg.pred)
                   
                   
rgd.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O", "OR", "OR", "OR", "OR", "YR", "YR")                 

ggbiplot(rgd.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(rgd.pca$x), groups=rgd.groups, varname.abbrev = TRUE)+ theme(legend.direction = 'horizontal', legend.position = 'top')+ggtitle("PCA Old vs Young, using just our subset of 61 markers + Regenerated samples projected")+ scale_colour_manual(values=c("cornflower blue","forest green", "red3",  "black"))


```

According to this, the regenerated samples are intermediate between the old and young samples. TYhis makes sense to me, as the OR samples being "younger" than the O samples fits the preliminary data, while the YR samples are three months older than the Y. 

#Marker expression levels in regenerated samples
But do these markers respond homogenously to regeneration? Lets look at how each marker responds to regeneration. 
We can plot the expression levels of each marker (normalised in relation to mean expression in old vs young samples) in old, young, regenerated young and regenerated old samples. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

mkcr <- cbind(mkc, bc[,c(5:6,17:20)]) 
nmkcr <- as.data.frame(mkcr[,c(1:15,32:37)]/rowMeans(as.matrix(mkcr[,c(1:15,32:37)])))
nmkcr <- cbind(as.data.frame(nmkcr), as.data.frame(mkc[,c(16:31)]))
nmkcr$rgl2fc <- foldchange2logratio(foldchange(rowMeans(mkcr[,c(1,6:9,14:15)]), rowMeans(mkcr[,c(32:35)])))
nmkcr$rgl2fc[nmkcr$dir == "upregulated with age"] <- nmkcr$rgl2fc[nmkcr$dir == "upregulated with age"]*-1
nmkcr$rgup <- nmkcr$rgl2fc >0
mnmkcr <- melt(nmkcr, id.vars=c(colnames(mkc[,c(16:31)]), "rgl2fc", "rgup"))
mnmkcr$class <- "Young"
mnmkcr$class[substr(mnmkcr$variable, 1,1) == "O"] <- "Old"
mnmkcr$class[substr(mnmkcr$variable, 1,2) == "OR"] <- "Old Regenerated"
mnmkcr$class[substr(mnmkcr$variable, 1,2) == "YR"] <- "Young Regenerated"
dm <- subset(mnmkcr, mnmkcr$p_value <0.05)
dm$dnm <- paste(substr(dm$dir, 1,2), dm$name)
l <- c("Younger with regen", "Older with regen")

ggplot(dm, aes(x=dnm, y=value, color=class, fill=dir)) +geom_point(shape=95, size=4.5)+stat_summary(fun.y=mean, geom="point", shape=21)+ scale_fill_manual(values=c("white", "azure4"), name="Direction")+ scale_y_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8) )+geom_hline(yintercept = 1, linetype=2)+scale_color_manual(values=c("cornflower blue","forest green", "red3",  "black"), name= "Age and Regeneration")+ theme(axis.text.x = element_text(angle = 60, hjust = 1))+ggtitle("61 chosen markers, normalised by mean expression")+ylab("Nanostring expression (normalised)") +xlab("Marker") +scale_x_discrete(labels=substr(dm$dnm, 4,5))
```

Some markers are upregulated with regeneration, some are downregulated with regeneration, irrespective of whether they are upregulated or downregulated with age. To see this more clearly, we will split those that move towards the young side (upper panel) from those that move towards the old side with regeneration. 


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

ggplot(dm, aes(x=dnm, y=value, color=class, fill=dir)) +geom_point(shape=95, size=4.5)+stat_summary(fun.y=mean, geom="point", shape=21)+ scale_fill_manual(values=c("white", "azure4"), name="Direction")+ scale_y_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8) )+geom_hline(yintercept = 1, linetype=2)+scale_color_manual(values=c("cornflower blue","forest green", "red3",  "black"), name= "Age and Regeneration")+ theme(axis.text.x = element_text(angle = 60, hjust = 1))+ggtitle("61 chosen markers, normalised by mean expression")+ylab("Nanostring expression (normalised)") +xlab("Marker")  +facet_wrap(~rgup, scales="free_x", ncol=1) + theme(strip.background = element_blank(), strip.text.x = element_blank())


```


So we have two sets of markers: those where the regenerated samples move towards the young condition (34 markers) and those where they move towards the old (27 markers)


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
#ggplot(subset(nmkcr, nmkcr$p_value <0.05), aes(x=name, y=rgl2fc)) +geom_point()
#ggplot(subset(nmkcr, nmkcr$p_value <0.05), aes(x=absnl2fc, y=rgl2fc)) +geom_point()

nmkcr$rgl2fc <- foldchange2logratio(foldchange(rowMeans(mkcr[,c(1,6:9,14:15)]), rowMeans(mkcr[,c(32:35)])))

#ggplot(nmkcr, aes(x=rgl2fc, fill=dir)) +geom_histogram(alpha=0.5) +scale_fill_manual(values=c())+theme(legend.position = "bottom", legend.title = element_blank()) +xlab("Log2Foldchange in expression in old vs regenerated old") +scale_fill_manual(name="x", values=c("darkcyan", "orangered")) 

#ggplot(nmkcr, aes(x=rgl2fc, fill=dir)) +geom_density(alpha=0.5) +scale_fill_manual(values=c())+theme(legend.position = "bottom", legend.title = element_blank()) +xlab("Log2Foldchange in expression in old vs regenerated old") +scale_fill_manual(name="x", values=c("darkcyan", "orangered")) 

```

Lets look at these on separate panels:

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

p<- ggplot(dm, aes(x=dnm, y=value, color=class, fill=dir)) +geom_point(shape=95, size=4.5)+theme_bw()+stat_summary(fun.y=mean, geom="point", shape=21)+ scale_fill_manual(values=c("white", "azure4"), name="Direction")+ scale_y_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8) )+scale_color_manual(values=c("cornflower blue","forest green", "red3",  "black"), name= "Age and Regeneration")+ theme(axis.text.x = element_text(angle = 60, hjust = 1))+ggtitle("61 chosen markers, normalised by mean expression")+ylab("Nanostring expression (normalised)") +xlab("Marker")  +facet_wrap(~class, scales="free_x", ncol=1) +geom_boxplot()+geom_hline(yintercept = 1, linetype=2)+scale_x_discrete(labels=substr(dm$dnm, 4,5)) 


ggsave(plot=p,height=10,width=10,dpi=200, filename=paste("t.pdf"), useDingbats=FALSE, limitsize = FALSE)


ggplot(dm, aes(x=dnm, y=value, color=class, fill=dir)) +geom_point(shape=95, size=4.5)+stat_summary(fun.y=mean, geom="point", shape=21)+ scale_fill_manual(values=c("white", "azure4"), name="Direction")+ scale_y_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8) )+scale_color_manual(values=c("cornflower blue","forest green", "red3",  "black"), name= "Age and Regeneration")+ theme(axis.text.x = element_text(angle = 60, hjust = 1))+ggtitle("61 chosen markers, normalised by mean expression")+ylab("Nanostring expression (normalised)") +xlab("Marker")  +facet_wrap(~class, scales="free_x", ncol=1) +geom_hline(yintercept = 1, linetype=2)+scale_x_discrete(labels=substr(dm$dnm, 4,5))

```

#Regeneration effect 

The old regenerated samples are much more variable than either the old or the young. Previously, we had split our markers into two groups: those where the regenerated samples move towards the young condition (34 markers) and those where they move towards the old (27 markers). Therefore, we might suspect that performing PCAs using these markers for young vs old will return different results. 


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

#nmkcr$rgup == FALSE & nmkcr$p_value <0.05)



bcg <- subset(bc, sig ==TRUE &nmkcr$rgup == FALSE)
bcg <- bcg[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcg)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
rd.pca <- bc.pca
rd<- bcg
rg <-subset(bc, sig ==TRUE &nmkcr$rgup == FALSE)
rg <- rg[,c(5:6,17:20)]


rg.sc <- scale(t(rg), center= rd.pca$center)
rg.pred <- rg.sc %*% rd.pca$rotation 


rgd.pca <- rd.pca
rgd.pca$x <- rbind(rd.pca$x, rg.pred)
                   
                   
rgd.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O", "OR", "OR", "OR", "OR", "YR", "YR")                 

pa <- ggbiplot(rgd.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(rgd.pca$x), groups=rgd.groups, varname.abbrev = TRUE)+ theme(legend.direction = 'horizontal', legend.position = 'top')+ggtitle("PCA Old vs Young, 34 markers + Regenerated samples projected")+ scale_colour_manual(values=c("cornflower blue","forest green", "red3",  "black"))



bcg <- subset(bc, sig ==TRUE &nmkcr$rgup == TRUE)
bcg <- bcg[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcg)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
rd.pca <- bc.pca
rd<- bcg
rg <-subset(bc, sig ==TRUE &nmkcr$rgup == TRUE)
rg <- rg[,c(5:6,17:20)]


rg.sc <- scale(t(rg), center= rd.pca$center)
rg.pred <- rg.sc %*% rd.pca$rotation 


rgd.pca <- rd.pca
rgd.pca$x <- rbind(rd.pca$x, rg.pred)
                   
                   
rgd.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O", "OR", "OR", "OR", "OR", "YR", "YR")                 

pb<- ggbiplot(rgd.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(rgd.pca$x), groups=rgd.groups, varname.abbrev = TRUE)+ theme(legend.direction = 'horizontal', legend.position = 'top')+ggtitle("PCA Old vs Young, 27 markers + Regenerated samples projected")+ scale_colour_manual(values=c("cornflower blue","forest green", "red3",  "black"))


multiplot(pa, pb, cols=1)
```



#Does regeneration have the same effect on old as on young?
Is logfoldchange in expression between old vs old regenerated correlated with logfoldchange in expression between young and young regenerated? 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

nmkcr$rgyl2fc <- foldchange2logratio(foldchange(rowMeans(mkcr[,c(2:5,10:13)]), rowMeans(mkcr[,c(36:37)])))
dn <- subset(nmkcr, nmkcr$p_value <0.05)

ggplot(dn, aes(x=rgyl2fc, y=rgl2fc, colour=dir)) +geom_point() +stat_smooth(aes(x=rgyl2fc, y=rgl2fc), method=lm, se=FALSE, inherit.aes = FALSE) +xlab("Log2FoldChange Young vs Regenerated Young") +ylab("Log2FoldChange Old vs Regenerated Old") +scale_colour_manual(values=c("darkcyan", "orangered"))+theme(legend.position = "bottom", legend.title = element_blank())
```

That looks to be an emphatic no!

#Nikos Plot

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

#View(nmkcr)
gm <- subset(nmkcr, nmkcr$sig == TRUE)

ggplot(gm, aes(x=nl2fc, y=rgyl2fc)) +geom_point()


hist(log(gm$OR1b))
hist(log(gm$OR2b))

dgm <- data.frame(gm$OR1a, gm$OR2a, gm$OR2a, gm$OR2b, gm$p_value, gm$dir, gm$rgup)
dgm <- melt(dgm, id.vars=c("gm.p_value", "gm.dir", "gm.rgup"))

ggplot(dgm, aes(x=value, fill=gm.rgup)) +geom_histogram(binwidth=0.25) +scale_x_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8)) +geom_point(aes(x=value, y=0, colour=gm.rgup))

ggplot(dgm, aes(x=value)) +geom_density() +scale_x_continuous(trans='log2',breaks=c(0.125,0.25,0.5,1,2,4), limits=c(0.125,8)) +geom_point(aes(x=value, y=0, colour=gm.dir)) 


+facet_wrap(~gm.dir)

```


#What effect do the Housekeeping genes have?

Let's look at our data without normalising via the housekeeping genes. Is our aging effect stable without doing this normalisation? 



```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

bcm <- subset(bcun, rownames(bcun) %in% rownames(bc))

oldmean <- rowMeans(bcm[,c(2,11,12,13,14,23,24)])
youngmean <- rowMeans(bcm[,c(3,4,9,10,15,16,21,22)])
OvYl2fc <- foldchange2logratio(foldchange(youngmean, oldmean))
nsOvY <- OvYl2fc

plot(OvYl2fc, codeset$log2FoldChange, xlab="Nanostring l2fc", ylab="Transcriptome l2fc", main= "New data only, all old vs all young, normalised by int controls only", pch=16, cex=0.8, col=adjustcolor(alpha=0.5, "dark blue")) +
lines(OvYl2fc[(OvYl2fc >1)&(codeset$log2FoldChange >0)], codeset$log2FoldChange[(OvYl2fc >1)&(codeset$log2FoldChange >0)], lty=0, col="forest green", type="p", pch=16, cex=0.8) + 
lines(OvYl2fc[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], codeset$log2FoldChange[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], lty=0, col="forest green", type="p", pch=16, cex=0.8) + 
rect(1,1,10,10, col=adjustcolor(alpha=0.2, "forest green"), lty=0)+ rect(-10,-10,-1,-1, col=adjustcolor(alpha=0.2, "forest green"), lty=0) + text(2,1, labels=paste(length(OvYl2fc[(OvYl2fc >1)&(codeset$log2FoldChange >0)&!is.na(OvYl2fc)]), "genes"), col="forest green") +grid()+ 
text(-2,-1, labels=paste(length(OvYl2fc[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)&!is.na(OvYl2fc)]), "genes"), col="forest green") +
text(OvYl2fc[(OvYl2fc >1)&(codeset$log2FoldChange >0)], codeset$log2FoldChange[(OvYl2fc >1)&(codeset$log2FoldChange >0)], labels=codeset$shortname[(OvYl2fc >1)&(codeset$log2FoldChange >0)], col="dark green", offset=0.5,1.3, cex=0.65) +
text(OvYl2fc[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], codeset$log2FoldChange[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], labels=codeset$shortname[(OvYl2fc+1 <0)&(codeset$log2FoldChange <0)], col="dark green", offset=0.5,1.3, cex=0.65)

```

Removing the normalisation doesn't seem to have had too much effect on the aging effect here. Let's look at the PCA. 

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
bcmin <- bcun[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 200 markers, normalised by internal controls only")+ scale_colour_manual(values=c("cornflower blue", "red3"))



```

We still return our old vs young separation, albeit split across two PCs. What about if we construct a PCA using just our housekeeping genes?

```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
bcmin <- subset(bcun, !(rownames(bcun) %in% rownames(bc)))
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=TRUE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 5 housekeeping genes")+ scale_colour_manual(values=c("cornflower blue", "red3"))



```

And what about the age effect seen when comparing old and young usign our 61 significant markers? Is that stable when we don't normalise via the housekeeping genes?


```{r, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}
bcmin <- subset(bcun, (rownames(bcun) %in% rownames(dbcg)))
bcmin <- bcmin[,c(2,3,4,9,10,11,12,13:16,21:24)]
tbc <- t(bcmin)
tbc <- tbc[,colSums(tbc) > 0]
bc.pca <- prcomp(tbc,center = TRUE,scale. = TRUE) 
bc.groups <- c("O", "Y", "Y", "Y", "Y", "O", "O", "O", "O", "Y", "Y", "Y", "Y", "O", "O")

ggbiplot(bc.pca, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = TRUE, var.axes=FALSE, labels=rownames(tbc), groups=bc.groups)+ theme(legend.direction = 'horizontal', legend.position = 'none')+ggtitle("PCA of O vs Y: 61 markers (normalised by internal controls only)")+ scale_colour_manual(values=c("cornflower blue", "red3"))



```

We still return the effect: our aging signature isn't an artefact of normalisation. This is heartening. 
